homeassistant/components/tuya/entity.py

```diff
@@ -177,5 +177,19 @@ async def _handle_state_update(
 
     def _send_command(self, commands: list[dict[str, Any]]) -> None:
         """Send command to the device."""
+        # Validate commands before sending
+        for command in commands:
+            if command.get("code") is None:
+                raise ValueError(
+                    f"Cannot send command with empty 'code' to device {self.device.id} "
+                    f"({self.device.name}). Command: {command}. This indicates the device "
+                    f"does not support the requested operation or Home Assistant cannot "
+                    f"find the correct control method. Available device capabilities: "
+                    f"function codes: {list(self.device.function.keys())}, status codes: "
+                    f"{list(self.device.status.keys())}. Your device may have limited "
+                    f"remote control support compared to what's available in the Smart Life "
+                    f"or Tuya Smart apps."
+                )
```

Comment:
When does this happen? Shouldn't we rather prevent these commands to be sent in the first place?

Reply:
Thanks for the reminder, this had moved into other verifications and I forgot to revert this. I'll remove it now.